lidar_localizer/nodes/ndt_matching/ndt_matching.cpp

```diff
@@ -1560,31 +1547,45 @@ int main(int argc, char** argv)
   private_nh.getParam("imu_upside_down", _imu_upside_down);
   private_nh.getParam("imu_topic", _imu_topic);
   private_nh.param<double>("gnss_reinit_fitness", _gnss_reinit_fitness, 500.0);
+  private_nh.getParam("output_tf_frame_id", _output_tf_frame_id);
```

Comment:
The way this is currently written, if this parameter is not provided the value of `_output_tf_frame_id` will be set to an empty string. To keep this backward-compatible, it should provide a default value in this call. Since you've already populated it with a default value prior to this call, this should work:

```
private_nh.getParam("output_tf_frame_id", _output_tf_frame_id, _output_tf_frame_id);
```

Reply:
I tested this and it doesn't seem to be the case, the docs aren't very clear as to what happens if the param doesn't exist, but it seems like it won't do anything to the variable.
Were you thinking of `ros::NodeHandle::paramros::NodeHandle::param`?

Reply:
You're right, the docs are unclear. I was going by my memory and apparently that isn't very clear either. Thanks for checking!